### ISSUE TITLE
Changing remote name causes publish to github modal 

### ISSUE THREAD

**Issue by mponce1** [github/VisualStudio#258]
Version: VS14 Update 1+
Steps to reproduce: 
1. Create and publish a repository
2. In repository settings, change the remote name to anything other than "origin"
3. Go to Sync

Expected result: Publish modal does not appear for repositories existing on dotcom. 
Actual result: Publish modal appears and shows an error for an existing branch.

Name set to "origin"
<img width="289" alt="screen shot 2016-04-05 at 10 00 08 am" src="https://cloud.githubusercontent.com/assets/6272408/14295149/47c562c0-fb29-11e5-9d80-fb6d88973773.png">

Name set to RTSServ2015
<img width="294" alt="screen shot 2016-04-05 at 11 23 24 am" src="https://cloud.githubusercontent.com/assets/6272408/14295166/54910d74-fb29-11e5-8f29-12386f4fe5c5.png">

**Comment by shana** [comment]
Yes, this is currently by design, we're only looking at the "origin" remote to check whether a repo is GitHub right now. Visual Studio itself is also doing only looking at remotes named "origin" (or it was until recently), need to check whether it's still doing it.

> Actual result: Publish modal appears and shows an error for an existing branch if the user clicks "get started" and attempts to publish.

s/existing branch/existing repository
